### PR TITLE
Use IWalletReceiver abstraction properly

### PR DESCRIPTION
**Pull request by omerfirmak**
This will allow us to build Outputs with different
LockTypes

### CODE REVIEW

File: src/modules/script/Lock.ts [hunk 1]
```diff
@@ -173,6 +173,14 @@ export class Lock {
     public clone(): Lock {
         return new Lock(this.type, this.bytes);
     }
+
+    /**
+     * Compare whether the two locks are the same.
+     * If it is the same, the true is returned. If different, false is returned.
```
Comment (hewison-chris) [inline src/modules/script/Lock.ts:179]:
```suggestion
     * If equal true, otherwise false is returned.
```

**Review by hewison-chris — APPROVED** [review]
LGTM

**Review by MichaelKim20 — APPROVED** [review]
LGTM

File: src/modules/script/Lock.ts (hunk 1 above)
Comment (omerfirmak) [reply src/modules/script/Lock.ts]: done

**Comment by omerfirmak** [comment]
Can we merge, I have more changes that depend on this.

**Pull request merged**